### PR TITLE
Update CI to include benchmarking changes in Test Suite.

### PR DESCRIPTION
**Pull request by saienduri**
This commit updates the SHARK-Test ref, config files, and yaml files to have the most up to date flags and benchmarking support. I will also concentrate on a python implementation for pulling in configs in Test-Suite, so we don't have to rely on all these config files. Checked the golden values over 15 times in the CI so should be good.

This commit also adds support to the CI, so that it generates a job summary of the benchmark mean times for e2e and all the sub models. This can be seen by developers in the summary tab of the PckgCI testing. Example: https://github.com/iree-org/iree/actions/runs/9501523985

image

Side note: The build_test_all_bazel was failing the first couple times and then passed. Seems to be unstable

### CODE REVIEW

File: build_tools/pkgci/external_test_suite/attention_and_matmul_spec.mlir [hunk 1]
```diff
@@ -469,6 +1054,38 @@ module attributes { transform.with_named_sequence } {
...
+        // Batch matmul.
+        , @match_batch_matmul_64x968x320x640 -> @apply_op_config
+        , @match_batch_matmul_64x968x640x640 -> @apply_op_config
+        , @match_batch_matmul_64x968x320x960 -> @apply_op_config
+        , @match_batch_matmul_64x242x1280x1280 -> @apply_op_config
+        , @match_batch_matmul_64x242x640x960 -> @apply_op_config
+        , @match_batch_matmul_64x242x640x1280 -> @apply_op_config
+        , @match_batch_matmul_64x242x640x1920 -> @apply_op_config
+
+        // Contration.
```
Comment (ScottTodd) [inline build_tools/pkgci/external_test_suite/attention_and_matmul_spec.mlir:1083]: Who maintains this spec file? I see some typos in here and a good portion of this file looks generated?

File: .github/workflows/pkgci_regression_test.yml [hunk 2]
```diff
@@ -254,10 +260,49 @@ jobs:
...
+            --goldentime-rocm-unet-ms 450.5 \
+            --goldentime-rocm-clip-ms 19 \
+            --goldentime-rocm-vae-ms 288.5 \
+            --gpu-number 6 \
+            --rocm-chip gfx90a \
+            --log-cli-level=info \
+            --retries 3
+          echo "### SDXL Benchmark Summary:" >> $GITHUB_STEP_SUMMARY
+          echo "" >> $GITHUB_STEP_SUMMARY # this is a blank line
+          echo "$(<job_summary.txt )" >> $GITHUB_STEP_SUMMARY
```
Comment (ScottTodd) [inline .github/workflows/pkgci_regression_test.yml:308]: Can you call this out in the PR description with an example link and/or screenshot? This is what we'd like contributors to be looking at when their changes affect performance.

File: .github/workflows/pkgci_regression_test.yml [hunk 3]
```diff
@@ -254,10 +260,49 @@ jobs:
             --log-cli-level=info \
             --timeout=1200 \
             --durations=0 \
-            --config-files=${SDXL_CONFIG_FILE_PATH}
+            --config-files=${SDXL_UNET_CONFIG_FILE_PATH}
+
+      - name: "Run external tests - SDXL prompt encoder"
+        if: "matrix.sdxl-clip-config-file != '' && !cancelled()"
```
Comment (ScottTodd) [inline .github/workflows/pkgci_regression_test.yml:266]: Let's continue keeping an eye on complexity here. New models shouldn't necessarily need workflow file changes.

1. Add imported model to the external test suite with .mlir, .irpa, and input/output .bin files
2. Bump test suite commit hash in this repo, choose which models from the test suite to run (on which backends, with which flags, etc.)

For these models I have the feeling that pytest (or at least this approach to pytest) is the wrong tool for the job. It's a tool that we have though, so fine for now...

File: build_tools/pkgci/external_test_suite/onnx_cpu_llvm_sync.json
Comment (ScottTodd) [inline build_tools/pkgci/external_test_suite/onnx_cpu_llvm_sync.json:1]: We should still move these files from `build_tools/pkgci/external_test_suite/` to somewhere under `tests/` IMO.

File: build_tools/pkgci/external_test_suite/attention_and_matmul_spec.mlir (hunk 1 above)
Comment (saienduri) [reply build_tools/pkgci/external_test_suite/attention_and_matmul_spec.mlir:1083]: Looks like it mostly deals with the attention and matmul ops along with tunings. Looks like a few people have been committing to this file here: https://github.com/nod-ai/sdxl-scripts/blob/main/specs/attention_and_matmul_spec.mlir.

File: .github/workflows/pkgci_regression_test.yml [hunk 4]
```diff
@@ -254,10 +260,49 @@ jobs:
...
-          bash SHARK-TestSuite/iree_tests/benchmarks/benchmark_sdxl_rocm.sh
+          pytest SHARK-TestSuite/iree_tests/benchmarks/benchmark_sdxl_rocm.py \
+            --goldentime-rocm-e2e-ms 1636 \
+            --goldentime-rocm-unet-ms 442 \
+            --goldentime-rocm-clip-ms 16.5 \
+            --goldentime-rocm-vae-ms 285 \
+            --gpu-number 3 \
+            --rocm-chip gfx90a \
+            --log-cli-level=info \
+            --retries 7
```
Comment (ScottTodd) [inline .github/workflows/pkgci_regression_test.yml:305]: oof, 7 retries x_x

that seems fairly arbitrary, so I expect we'll be tuning the thresholds, benchmark iterations, and retry count ...

**Review by ScottTodd — APPROVED** [review]

File: .github/workflows/pkgci_regression_test.yml (hunk 4 above)
Comment (saienduri) [reply .github/workflows/pkgci_regression_test.yml:305]: Yup, that was for when I temporarily changed the unet to 441. Thought it wouldn't hurt to leave it at 7, but in the testing I did, never got there with these values. But, yeah we can tune these numbers as we go :)

**Pull request merged**